core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
> If this order is completed, I expect the promotion to be ineligible.

Yeah the only case I can think of is if an admin were retroactively applying a promotion after an order was completed. John's idea of counting 'orders completed before this order' might cover even that case though.